changelogs/unreleased/fix-replica-disconnect-memleak.md

```diff
@@ -0,0 +1,3 @@
+## bugfix/replication
+
+- Fixed a memory leak on disconnection from replica.
```

Comment:
I think this bug can leak quite a lot of memory if a replica connects and disconnects repeatedly due to some kind of error or poor connectivity, can't it? Please consider filing an issue for this one, as well.

Reply:
Created an issue #10480